### PR TITLE
invite codes and list federations even if you're in one

### PR DESCRIPTION
**Pull request by futurepaul**
@TonyGiorgio and I discussed this over steak tartare

Screenshot 2024-04-12 at 11 41 17 AM

Screenshot 2024-04-12 at 11 41 21 AM

### CODE REVIEW

**Comment by benthecarman** [comment]
If you guys want this that is fine with me, but curious what is the reasoning for this?

File: src/components/FederationInviteShower.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            <div class="flex w-full justify-center gap-8">
+                <button onClick={() => setShowQr(true)}>
+                    <QrCode class="inline-block h-4 w-4" />
+                </button>
+                <button
+                    class="p-1"
+                    classList={{
+                        "bg-m-red rounded": copied()
+                    }}
+                    onClick={() => copy(props.inviteCodes[0])}
```
Comment (benthecarman) [inline src/components/FederationInviteShower.tsx:29]: Would be cool if you could scroll through the invite codes, in case the first isn't working

**Comment by futurepaul** [comment]
> If you guys want this that is fine with me, but curious what is the reasoning for this?

it's just nice for reasons of curiosity imo

File: src/components/FederationInviteShower.tsx (hunk 1 above)
Comment (futurepaul) [reply src/components/FederationInviteShower.tsx]: we don't actually have multiple. made it singular

**Review by benthecarman — COMMENTED** [review]
when you remove a federation it reloads the list still, shoulnd't need to do that anymore but not the end of the world

**Comment by futurepaul** [comment]
> when you remove a federation it reloads the list still

yeah it's a lot simpler code-wise this way

**Review by benthecarman — APPROVED** [review]

**Comment by TonyGiorgio** [comment]
> If you guys want this that is fine with me, but curious what is the reasoning for this?

multiple people are asking me for it, they want to know the federations that exist.

**Pull request merged**